Permanent/ViewControllers/TagDetailsViewController.swift

```diff
@@ -0,0 +1,260 @@
+            self.hideSpinner()
+            self.delegate?.tagDetailsViewControllerDidUpdate(self)
+            self.dismiss(animated: true, completion: nil)
+        }
+
+    }
+    
+    @IBAction func AddTagButtonAction(_ sender: Any) {
+        var tagsList: [String] = []
+        
+        for item in sortedArray {
+            if let itemName = item.tagVO.tagVO.name {
+                tagsList.append(itemName)
+            }
+        }
+        
+        if let findTag = tagFindSearchBar.text,
+           !findTag.isEmpty,
+           !tagsList.contains(findTag)
+        {
+            sortedArray.insert(SortedTagVO(tagVO: TagVO(tagVO: TagVOData(name: findTag, status: String(), tagId: Int(), type: String(), createdDT: String(), updatedDT: String())), checked: true, forAdding: true, forRemoval: false), at: 0)
+            
+            self.tagFindSearchBar.text = ""
+            
+            filteredTagVO = sortedArray.map({ (item) -> TagVO in
+                return item.tagVO
+            })
+            
+            self.tagsCollectionView.reloadData()
+            self.view.endEditing(true)
+        }
+    }
+    
+    func initTagsState()
+    {
+        showSpinner(colored: .lightGray)
+        viewModel?.getTagsByArchive(archiveId: viewModel?.file.archiveId ?? 0, completion: { result in
+            guard let tagsArchive = result else {
+                return
+            }
+            
+            for (index,_) in tagsArchive.enumerated() {
+                self.sortedArray.append(SortedTagVO(tagVO: tagsArchive[index]))
+                
+            }
+            
+            if let tags = self.tagVOS {
+                for tagItem in tags {
+                    var idx : Int?
+                    idx = self.sortedArray.firstIndex(where: { $0.tagVO.tagVO.name == tagItem.name})
+
+                    if idx != nil { self.sortedArray[idx!].checked = true }
+                }
+            }
+            self.sortedArray.sort { (_, arg0) -> Bool in !arg0.checked }
+            self.filteredTagVO = self.sortedArray.map({ (item) -> TagVO in
+                return item.tagVO
+            })
+
+            self.tagsCollectionView.reloadData()
+            self.hideSpinner()
+        })
+    }
+}
+
+extension TagDetailsViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return filteredTagVO.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: TagCollectionViewCell.identifier, for: indexPath) as! TagCollectionViewCell
+        
+        if let tagName = filteredTagVO[indexPath.row].tagVO.name {
+            cell.configure(name: tagName, isVisible: sortedArray[indexPath.row].checked)
```

Comment:
shouldn't isVisible come from filteredTagVO array as well?

Reply:
if let tagName = filteredTagVO[indexPath.row].tagVO.name,
           let isChecked = sortedArray.first(where: { $0.tagVO.tagVO.name == tagName })?.checked {
            cell.configure(name: tagName, isVisible: isChecked)
        }

Reply:
Adapted logic to use isChecked variable.